k9mail-library/src/main/java/com/fsck/k9/mail/store/imap/ImapFolderPusher.java

```diff
@@ -253,6 +253,8 @@ private void reacquireWakeLockAndCleanUp() {
             } catch (Exception me) {
                 Log.e(LOG_TAG, "Got exception while closing for exception for " + getLogId(), me);
             }
+
+            connection = null;
```

Comment:
Can you describe a situation where `connection` would be `null` when entering this method?